common-jvm/src/main/java/org/ergoplatform/ErgoFacade.kt

```diff
@@ -286,14 +287,41 @@ fun signMessage(
     }
 }
 
-private fun getRestErgoClient(prefs: PreferencesProvider) =
-    RestApiErgoClient.createWithHttpClientBuilder(
-        prefs.prefNodeUrl,
+private fun getRestErgoClient(prefs: PreferencesProvider): ErgoClient {
+    val nodeToConnectTo = prefs.prefNodeUrl
+    val ergoClient = RestApiErgoClient.createWithHttpClientBuilder(
+        nodeToConnectTo,
         getErgoNetworkType(),
         "",
         prefs.prefExplorerApiUrl,
         OkHttpSingleton.getInstance().newBuilder()
     )
+    refreshNodeListWhenNeeded(prefs, nodeToConnectTo)
+    return ergoClient
+}
+
+private fun refreshNodeListWhenNeeded(prefs: PreferencesProvider, nodeUrl: String) {
+    if (System.currentTimeMillis() - prefs.lastNodeListRefreshMs > 1000L * 60 * 60 * 24) {
+        val retrofitNode = Retrofit.Builder()
+            .baseUrl(nodeUrl)
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(OkHttpSingleton.getInstance())
+            .build()
+        val peersApi = retrofitNode.create(PeersApi::class.java)
+
+        try {
+            val peerUrlList = peersApi.connectedPeers.execute().body()!!.map { it.restApiUrl }
+            val openPeers = peerUrlList.filter { !it.isNullOrBlank() && it != nodeUrl }.distinct()
+            val peersStringList = openPeers.sortedBy { if (it.startsWith("https://")) 0 else 1 }.take(10)
+
+            LogUtils.logDebug("PeersApi", "Found alternative nodes to connect to: $peersStringList")
+            if (peersStringList.isNotEmpty())
+                prefs.knownNodesList = peersStringList
```

Comment:
The behaviour is the following so far:
The node that is connected is always the one set by the user. This should stay this way because in theory nodes can be malicious or unreliable, an automatic change should not be made.

So the nodes saved in this lists are a buffer for when the node that is connected to is down. In this case, the user can press the wizard icon in the settings screen to set another node.

When the node has only a single peer in connectedPeers but we had more in the list before, then it means the node removed the peers by some reason and it is good to have them removed from the list as well.

The behaviour is still not perfect and users need to make a manual step in case there are node problems, that's why the error message contains a hint now. I see too much risks with the automatic change.